pkg/core/codeset_store.go

```diff
@@ -4,6 +4,7 @@ import (
 	"github.com/pkg/errors"
 
 	codeset "github.com/fuseml/fuseml-core/gen/codeset"
+	giteaadmin "github.com/fuseml/fuseml-core/pkg/core/gitea"
```

Comment:
For the dependency inversion to be fully effective, I would say you shouldn't need to add any dependency here to the gitea package. The quote goes: _High-level modules should not depend on low-level modules. Both should depend on abstractions (e.g. interfaces)_. In this case I think that means the codeset store shouldn't depend on anything from `core/gitea`, it should only import the interface from `core/gitadmin`.

Reply:
I don't think so. There needs to be some code that actually imports the file where the implementation is happening. To quote some other part of that article,

 _`main` is the only package which needs to be aware of those 2 components_. 

(`codeset_store.go` is our `main` here)

Reply:
I think that was meant literally, and `fuseml_core/main` is our main package :)

Reply:
No, I do not think so, in that example it's `main` package which calls the storage and foo parts, that's why I'm saying `codeset_store.go` is our `main` compared to this example

Reply:
My point is, _someone_ needs to instantiate NewGiteaAdminClient before it can be passed to interface. And you just can't do it without some imports.

Anyway, maybe I'm missing something, this golang-specific concept is still confusing to me...